Formula/c/cppinsights.rb

```diff
@@ -16,18 +17,25 @@ class Cppinsights < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "llvm"
+  depends_on "llvm@20"
 
-  fails_with :clang do
-    build 1500
-    cause "Requires Clang > 15.0"
-  end
+  # TODO: Restore compiler selection when using unversioned `llvm`
+  # fails_with :clang do
+  #   build 1500
+  #   cause "Requires Clang > 15.0"
+  # end
 
   def install
-    ENV.llvm_clang if OS.mac? && DevelopmentTools.clang_build_version <= 1500
+    llvm = Formula["llvm@20"]
+    if OS.mac? && DevelopmentTools.clang_build_version <= 1500
+      # TODO: Restore ENV.llvm_clang when using unversioned `llvm`
+      ENV["CC"] = llvm.opt_bin/"clang"
+      ENV["CXX"] = llvm.opt_bin/"clang++"
+      ENV.append_to_cflags "-Wno-old-style-cast" # try to remove when using shim
```

Comment:
Will continue this in a follow up PR. Likely need to drop one of the `-W` from https://github.com/andreasfertig/cppinsights/blob/main/CMakeLists.txt#L293-L295
```cmake
        add_definitions(-Wold-style-cast)
        if(NOT DEBUG)
            add_definitions(-Werror)
```